lib/core/uri_parser.js

```diff
@@ -551,8 +555,8 @@ function parseConnectionString(uri, options, callback) {
       return callback(new MongoParseError('Unescaped colon in authority section'));
     }
 
-    auth.username = qs.unescape(authParts[0]);
-    auth.password = authParts[1] ? qs.unescape(authParts[1]) : null;
+    if (!auth.username) auth.username = qs.unescape(authParts[0]);
```

Comment:
When would this apply?

Reply:
If auth options are specified via an options object, like on lines 532-540, then we should respect those before we respect the values in the url string. I believe in general our rule is to respect manually-set options before url string options.